clusterspec/clusterspec.go

```diff
@@ -0,0 +1,133 @@
+package clusterspec
+
+import (
+	"net/url"
+	"strings"
+
+	v1alpha1core "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	v1alpha1provider "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+)
+
+// Config holds required configuration values for factory to be able to
+// construct provider independent Cluster configuration based on
+// ClusterGuestConfig.
+type Config struct {
+	Calico struct {
+		CIDR   int
+		MTU    int
+		Subnet string
+	}
```

Comment:
I am not a fan of inline structures. They are weird to work with. There is also already this common cluster type in the apiextensions repo. Why aren't we just using this? Repeating all the jazz here is not necessary IMO.

Reply:
I'm fine with splitting this to separate structures. I'll take a look at replacing this with `Cluster` from apiextensions, but initially I get a feeling this does not make sense: Having factory configured with type that it is built to produce feels wrong.

Reply:
I think when this is all configured via a viper then all this comes from a configmap in our case and because we are using microkit we have this flag package structure so we should maybe just copy the whole thing and use it here again to have it straight.

Reply:
That sounds like a good idea. I will investigate this further.